tests/lcb/stdlib/arithmetic.lcb

```diff
@@ -0,0 +1,210 @@
+/*
+Copyright (C) 2015 Runtime Revolution Ltd.
+
+This file is part of LiveCode.
+
+LiveCode is free software; you can redistribute it and/or modify it under
+the terms of the GNU General Public License v3 as published by the Free
+Software Foundation.
+
+LiveCode is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received a copy of the GNU General Public License
+along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
+
+module com.livecode.arithmetic.tests
+
+public handler TestUnarySigns()
+	variable t
+	put 10 into t
+	test "unary−" when -t is -10
+
+	put -10 into t
+	test "unary+" when +t is -10
+end handler
+
+public handler TestAdd()
+	variable tInt as integer
+	put 10 into tInt
+
+	add 2 to tInt
+	test "add (syntax, int)" when tInt = 12
+	test "add (operator, int)" when tInt + 1 = 13
+
+	add 1.9 to tInt
+	test diagnostic "Expecting an error"
+	broken test "add (syntax, int+real)" when tInt = 13 because "bug 14592"
+
+	variable tReal as real
+	put 10 into tReal
+	add 1.9 to tReal
+
+	test "add (syntax, real)" when tReal = 11.9
+	test "add (operator, real)" when tReal + 0.2 = 12.1
+
+	variable tNumber
+	put 10 into tNumber
+	add -1.9 to tNumber
+
+	test "add (syntax, number)" when tNumber = 8.1
+	test "add (operator, number)" when tNumber + -1 = 7.1
+end handler
+
+public handler TestSubtract()
+	variable tInt as integer
+	put 10 into tInt
+
+	subtract 2 from tInt
+	test "subtract (syntax, int)" when tInt = 8
+	test "subtract (operator, int)" when tInt - -2 = 10
+
+	subtract 1.9 from tInt
+	test diagnostic "Expecting an error"
+	broken test "subtract (syntax, int-real)" when tInt = 6 because "bug 14592"
+
+	variable tReal as real
+	put 10 into tReal
+	subtract 1.9 from tReal
+
+	test "subtract (syntax, real)" when tReal = 8.1
+	test "subtract (operator, real)" when tReal - 0.25 = 7.85
+
+	variable tNumber
+	put 10 into tNumber
+	subtract -1.9 from tNumber
+
+	test "subtract (syntax, number)" when tNumber = 11.9
+	test "subtract (operator, number)" when tNumber - 1 = 10.9
+end handler
+
+public handler TestMultiply()
+	variable tInt as integer
+	put 10 into tInt
+
+	multiply tInt by 2
+	test "multiply (syntax, int)" when tInt = 20
+	test "multiply (operator, int)" when tInt * 2 = 40
+
+	variable tReal as real
+	put 10 into tReal
+	multiply tReal by -0.5
+	test "multiply (syntax, real)" when tReal = -5
+	test "multiply (operator, real)" when tReal * 3.5 = -17.5
+
+	variable tNumber
+	put 10 into tNumber
+	multiply tNumber by 0.5
+	test "multiply (syntax, number)" when tNumber = 5
+	test "multiply (operator, number)" when tNumber * 3.5 = 17.5
+end handler
+
+public handler TestDivide()
+	variable tInt as integer
+	put 10 into tInt
+
+	divide tInt by 2
```

Comment:
I've made the 'divide' syntax and '/' operator always produce reals - if you want integer division you have to use div. (This is consistent with LCS).